Incorrect build context for build-api

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Dify version

main

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

Build docker container from the repository root with the following command:

```bash
make build-api -e API_IMAGE=ghcr.io/quantumghost/dify -e VERSION=main
```

### ✔️ Expected Behavior

Docker build succeeded without any issue.

### ❌ Actual Behavior

Docker build failed with the following error:

```
Building API Docker image: ghcr.io/quantumghost/dify:main...
docker build -t ghcr.io/quantumghost/dify:main ./api
[+] Building 1.0s (20/20) FINISHED                                                                                                                                 docker:dev-hk
 => [internal] load build definition from Dockerfile                                                                                                                        0.5s
 => => transferring dockerfile: 4.48kB                                                                                                                                      0.4s
 => WARN: SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ARG "NODESOURCE_KEY_FPR") (line 59)                                                 0.5s
 => [internal] load metadata for docker.io/library/python:3.12-slim-bookworm                                                                                                0.0s
 => [internal] load build context                                                                                                                                           0.2s
 => => transferring context: 2B                                                                                                                                             0.2s
 => [base 1/3] FROM docker.io/library/python:3.12-slim-bookworm@sha256:93ab4b7fa528b25124c97bcc755415e60eb671a86b4dbe0328df2fe2d1c1193d                                     0.0s
 => => resolve docker.io/library/python:3.12-slim-bookworm@sha256:93ab4b7fa528b25124c97bcc755415e60eb671a86b4dbe0328df2fe2d1c1193d                                          0.0s
 => CACHED [base 2/3] WORKDIR /app/api                                                                                                                                      0.0s
 => CACHED [base 3/3] RUN pip install --no-cache-dir uv==0.8.9                                                                                                              0.0s
 => CACHED [packages 1/6] RUN apt-get update     && apt-get install -y --no-install-recommends           git g++           libmpfr-dev libmpc-dev                           0.0s
 => ERROR [packages 2/6] COPY api/pyproject.toml api/uv.lock ./                                                                                                             0.0s
 => ERROR [packages 3/6] COPY api/providers ./providers                                                                                                                     0.0s
 => CACHED [production 1/8] WORKDIR /app/api                                                                                                                                0.0s
 => CACHED [production 2/8] RUN groupadd -r -g 1001 dify &&     useradd -r -u 1001 -g 1001 -s /bin/bash dify &&     chown -R dify:dify /app                                 0.0s
 => CACHED [production 3/8] RUN     apt-get update     && apt-get install -y --no-install-recommends         ca-certificates         curl         gnupg     && mkdir -p /e  0.0s
 => ERROR [packages 4/6] COPY dify-agent/pyproject.toml dify-agent/README.md /app/dify-agent/                                                                               0.0s
 => ERROR [packages 5/6] COPY dify-agent/src /app/dify-agent/src                                                                                                            0.0s
 => CACHED [packages 6/6] RUN uv sync --frozen --no-dev --no-editable                                                                                                       0.0s
 => CACHED [production 4/8] COPY --from=packages --chown=dify:dify /app/api/.venv /app/api/.venv                                                                            0.0s
 => CACHED [production 5/8] RUN mkdir -p /usr/local/share/nltk_data     && NLTK_DATA=/usr/local/share/nltk_data python -c "import nltk; nltk.download('punkt'); nltk.downl  0.0s
 => CACHED [production 6/8] RUN python -c "import tiktoken; tiktoken.encoding_for_model('gpt2')"     && chown -R dify:dify /app/api/.tiktoken_cache                         0.0s
 => ERROR [production 7/8] COPY --chown=dify:dify api /app/api/                                                                                                             0.0s
 => ERROR [production 8/8] COPY --chown=dify:dify --chmod=755 api/docker/entrypoint.sh /entrypoint.sh                                                                       0.0s
------
 > [packages 2/6] COPY api/pyproject.toml api/uv.lock ./:
------
------
 > [packages 3/6] COPY api/providers ./providers:
------
------
 > [packages 4/6] COPY dify-agent/pyproject.toml dify-agent/README.md /app/dify-agent/:
------
------
 > [packages 5/6] COPY dify-agent/src /app/dify-agent/src:
------
------
 > [production 7/8] COPY --chown=dify:dify api /app/api/:
------
------
 > [production 8/8] COPY --chown=dify:dify --chmod=755 api/docker/entrypoint.sh /entrypoint.sh:
------

 1 warning found (use docker --debug to expand):
 - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ARG "NODESOURCE_KEY_FPR") (line 59)
Dockerfile:116
--------------------
 114 |     
 115 |     # Prepare entrypoint script
 116 | >>> COPY --chown=dify:dify --chmod=755 api/docker/entrypoint.sh /entrypoint.sh
 117 |     
 118 |     
--------------------
ERROR: failed to build: failed to solve: failed to compute cache key: failed to calculate checksum of ref r9ml19wmb4vdza2birpkr980o::rfbs9ciwyw0dobsu51jb7kca1: "/api/docker/entrypoint.sh": not found
make: *** [Makefile:160: build-api] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -157,7 +157,7 @@ build-web:
 
 build-api:
 	@echo "Building API Docker image: $(API_IMAGE):$(VERSION)..."
-	docker build -t $(API_IMAGE):$(VERSION) ./api
+	docker build -t $(API_IMAGE):$(VERSION) -f api/Dockerfile .
 	@echo "API Docker image built successfully: $(API_IMAGE):$(VERSION)"
 
 # Push Docker images

--- a/api/Dockerfile.dockerignore
+++ b/api/Dockerfile.dockerignore
@@ -8,18 +8,30 @@
 !dify-agent/src/
 !dify-agent/src/**
 
-api/.venv
-api/.venv/**
-api/.env
-api/*.env.*
-api/.idea
-api/.mypy_cache
-api/.ruff_cache
-api/storage/generate_files/*
-api/storage/privkeys/*
-api/storage/tools/*
-api/storage/upload_files/*
-api/logs
-api/*.log*
+# Environment configuration and example
+.env
+*.env.*
+
+# Python related files
 **/__pycache__
 **/*.pyc
+**/.venv/
+**/.mypy_cache/
+**/.ruff_cache/
+**/.import_linter_cache/
+**/.pytest_cache/
+**/.hypothesis/
+
+
+# Upload files and logs
+api/storage/**
+api/logs/
+api/*.log*
+
+# Tests
+api/tests
+
+
+# Editor configuration
+**/.vscode/
+**/.idea/

